### PR TITLE
Add send function to new UI

### PR DESCRIPTION
**Pull request by jubalm**
This only does transfer, the Transaction details view will be on a separate PR #115

### CODE REVIEW

File: app/ts/components/AmountField.tsx [hunk 1]
```diff
@@ -1,40 +1,61 @@
...
+	const handleClear = () => {
+		props.onClear?.()
+		inputRef.current?.focus()
+	}
+
+	const handleInput = (e: JSX.TargetedEvent<HTMLInputElement>) => {
+		const inputField = inputRef.current
+		const value = e.currentTarget.value
+
+		if (inputField === null) return;
```
Comment (KillariDev) [inline app/ts/components/AmountField.tsx:29]:
```suggestion
		if (inputField === null) return
```

File: app/ts/components/TransferPage/index.tsx [hunk 2]
```diff
@@ -63,12 +65,12 @@ const MainPanel = () => {
...
 						<div class='grid md:grid-cols-2 gap-4'>
-							<TokenField token={formData.value.token} onClick={() => tokenManager.value = true} />
-							<InputField label='Amount' placeholder='1.00' value={formData.value.amount} onInput={(value) => formData.value = { ...formData.value, amount: value }} onClear={() => formData.value = { ...formData.value, amount: '' }} />
+							<TokenField token={data.value.token} onClick={() => tokenManager.value = true} disabled={isFormSubmitting.value} />
+							<AmountField label='Amount' placeholder='1.00' value={data.value.amount} onInput={(value) => data.value = { ...data.value, amount: value }} onClear={() => data.value = { ...data.value, amount: '' }} disabled={isFormSubmitting.value} />
 						</div>
-
-						<InputField label='Addrses' placeholder='0x123...789' value={formData.value.address} onInput={handleAddressChange} onClear={handleAddressChange} />
-						<SubmitButton />
+						<AddressField label='Addrses' placeholder='0x123...789' value={data.value.recipientAddress} onInput={handleAddressChange} onClear={handleAddressChange} disabled={isFormSubmitting.value} />
```
Comment (KillariDev) [inline app/ts/components/TransferPage/index.tsx:71]:
```suggestion
						<AddressField label='address' placeholder='0x123...789' value={data.value.recipientAddress} onInput={handleAddressChange} onClear={handleAddressChange} disabled={isFormSubmitting.value} />
```

File: app/ts/components/TransferPage/index.tsx [hunk 3]
```diff
@@ -152,49 +150,58 @@ const TokenField = (props: TokenFieldProps) => {
...
-
-			</div>
-		</div>
-
-	)
+			)
+		case 'resolved':
+			return <></>
+		case 'rejected':
+			console.log(typeof transaction.value.error)
```
Comment (KillariDev) [inline app/ts/components/TransferPage/index.tsx:196]: left over debug print?

File: app/ts/components/TransferPage/index.tsx [hunk 4]
```diff
@@ -152,49 +150,58 @@ const TokenField = (props: TokenFieldProps) => {
...
+			return (
+				<div class='grid gap-2 border border-red-400/50 px-4 py-3 bg-red-200/10'>
+					<div class='font-bold text-lg'>Failed to complete transfer</div>
+					<div>Your connected wallet returned with the following message</div>
```
Comment (KillariDev) [inline app/ts/components/TransferPage/index.tsx:200]:
```suggestion
					<div>Your wallet returned with the following message</div>
```

File: app/css/index.css [hunk 5]
```diff
@@ -1493,6 +1505,12 @@ img,
   transition-duration: 300ms;
 }
 
+.clear:after,
+	.clear:before {
+  content: ' ';
```
[the remark this answers is not in the document]
Comment (KillariDev) [reply app/css/index.css:1510]: These CSS files are using spaces and the other files tabs. These could be unified I think?

File: app/ts/store/transfer.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+import { useProviders } from './provider.js'
+import { TokenMeta } from './tokens.js'
+
+type TransferData = {
+	recipientAddress: string,
+	amount: string,
+	token?: TokenMeta
+}
+
+const transferDataDefaults:TransferData = { recipientAddress: '', amount: '', token: undefined }
```
Comment (KillariDev) [inline app/ts/store/transfer.ts:15]:
```suggestion
const transferDataDefaults: TransferData = { recipientAddress: '', amount: '', token: undefined }
```

**Review by KillariDev — APPROVED** [review]

File: app/css/index.css (hunk 5 above)
[the remark this answers is not in the document]
Comment (jubalm) [reply app/css/index.css]: removed the `.clear` utilities as they are no longer used. furthermore, css is auto-generated by tailwind at the moment so I'm leaving it defaults for now until we need to optimize.

**Pull request merged**